cmd/create/create_agent.go

```diff
@@ -0,0 +1,439 @@
+	}
+
+	ctx, cancel := context.WithTimeout(o.Context, time.Minute)
+	defer cancel()
+
+	// fetch the list of agents
+	existingAgents, err := o.ArmoryClient.Agents().List(ctx)
+	if err != nil {
+		return err
+	}
+
+	agentNameAlreadyExistFunc := func(name string) bool {
+		_, exists := lo.Find(existingAgents, func(a model.Agent) bool {
+			return name == a.AgentIdentifier
+		})
+		return exists
+	}
+
+	// set agent name
+	promptSetAgentName := promptui.Prompt{
+		Label: fmt.Sprintf("Provide an agent identifier %s", lo.Ternary(agentNameAlreadyExistFunc(requestedContext), "", fmt.Sprintf("[default=%s]", requestedContext))),
+		Validate: func(name string) error {
+			if agentNameAlreadyExistFunc(name) {
+				return errors.New("sorry, there's already an agent with that name in your tenant")
+			}
+			return nil
+		},
+	}
+
+	agentName, err := promptSetAgentName.Run()
+	if err != nil {
+		return errors.New(fmt.Sprintf("failed to set the agent name; %v\n", err))
+	}
+
+	if lo.IsEmpty(agentName) {
+		agentName = requestedContext
+	}
+
+	o.Name = agentName
+
+	// set namespace
+	promptSetNamespace := promptui.Prompt{
+		Label:  fmt.Sprintf("Provide a namespace where the agent will be installed [default=%s]", defaultNamespaceName),
+		Stdout: &util.BellSkipper{},
+	}
+
+	namespaceName, err := promptSetNamespace.Run()
+	if err != nil {
+		return errors.New(fmt.Sprintf("failed to set the namespace; %v\n", err))
+	}
+
+	if lo.IsEmpty(namespaceName) {
+		namespaceName = defaultNamespaceName
+	}
+
+	o.Namespace = namespaceName
+
+	// fetch the list of credentials
+	existingCredentials, err := o.ArmoryClient.Credentials().List(ctx)
+	if err != nil {
+		return err
+	}
+
+	// create new credentials
+	credentials := o.createCredentials()
+	existingCredential, credentialsExists := lo.Find(existingCredentials, func(c *model.Credential) bool {
+		return credentials.Name == c.Name
+	})
+
+	if credentialsExists {
+		// recreate credentials
+		promptRecreateCredentials := promptui.Prompt{
+			Label:     fmt.Sprintf("A Client Credential named %s already exists. Do you want to generate a new Client Credentials?", o.Name),
+			IsConfirm: true,
+			Default:   "Y",
+			Stdout:    &util.BellSkipper{},
+		}
+
+		if _, err := promptRecreateCredentials.Run(); err != nil {
+			return errors.New(fmt.Sprintf("Exiting %s\n", err))
+		}
+
+		err = o.ArmoryClient.Credentials().Delete(ctx, existingCredential)
+		if err != nil {
+			return errors.New(fmt.Sprintf("failed to delete credentials; %v\n", err))
+		}
+	}
+
+	credentials, err = o.ArmoryClient.Credentials().Create(ctx, credentials)
+	if err != nil {
+		return err
+	}
+
+	existingRoles, err := o.ArmoryClient.Roles().ListForMachinePrincipals(ctx, o.configuration.GetCustomerEnvironmentId())
+	if err != nil {
+		return err
+	}
+
+	// add the RNA role to the newly created credentials
+	rol, rolExists := lo.Find(existingRoles, func(c model.RoleConfig) bool {
+		return "Remote Network Agent" == c.Name
```

Comment:
I wouldn't want to depend on this string. I would check for a role that
1. has the `systemDefined` boolean `true`
2. has the `api:agentHub:full` scope (using the constants from `go-commons`)

(Some combination of these constants: https://github.com/armory-io/go-commons/blob/main/iam/scopes/scopes.go#L15-L39)

Reply:
I guess you might have to fetch the credential's roles to do that - I think that's worth it.